src/Raven.Server/Documents/Queries/Suggestions/Suggestions.cs

```diff
@@ -0,0 +1,757 @@
+﻿/* 
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System.Collections.Generic;
+using System.Threading;
+using Lucene.Net.Util;
+
+namespace SpellChecker.Net.Search.Spell
+{
+    using System;
+
+    using Lucene.Net.Search;
+    using Lucene.Net.Store;
+    using BooleanClause = Lucene.Net.Search.BooleanClause;
+    using BooleanQuery = Lucene.Net.Search.BooleanQuery;
+    using Directory = Lucene.Net.Store.Directory;
+    using Document = Lucene.Net.Documents.Document;
+    using Field = Lucene.Net.Documents.Field;
+    using IndexReader = Lucene.Net.Index.IndexReader;
+    using IndexSearcher = Lucene.Net.Search.IndexSearcher;
+    using IndexWriter = Lucene.Net.Index.IndexWriter;
+    using Query = Lucene.Net.Search.Query;
+    using Term = Lucene.Net.Index.Term;
+    using TermQuery = Lucene.Net.Search.TermQuery;
+    using WhitespaceAnalyzer = Lucene.Net.Analysis.WhitespaceAnalyzer;
+
+
+    /// <summary>  <p>
+    /// Spell Checker class  (Main class) <br/>
+    /// (initially inspired by the David Spencer code).
+    /// </p>
+    /// 
+    /// <p>Example Usage:</p>
+    /// 
+    /// <pre>
+    /// SpellChecker spellchecker = new SpellChecker(spellIndexDirectory);
+    /// // To index a field of a user index:
+    /// spellchecker.indexDictionary(new LuceneDictionary(my_lucene_reader, a_field));
+    /// // To index a file containing words:
+    /// spellchecker.indexDictionary(new PlainTextDictionary(new File("myfile.txt")));
+    /// string[] suggestions = spellchecker.suggestSimilar("misspelt", 5);
+    /// </pre>
+    /// 
+    /// </summary>
+    /// <author>  Nicolas Maisonneuve
+    /// </author>
+    /// <version>  1.0
+    /// </version>
+    internal sealed class SpellChecker : IDisposable
```

Comment:
It didnt changed much from the current Lucene code, this is the oldest model available from Lucene, so current code is up-to-date with Java Lucene.